shell/browser/ui/file_dialog_gtk.cc

```diff
@@ -131,24 +131,20 @@ class FileChooserDialog {
       : parent_(
             static_cast<electron::NativeWindowViews*>(settings.parent_window)),
         filters_(settings.filters) {
-    const char* confirm_text = gtk_util::kOkLabel;
-
-    if (!settings.button_label.empty())
-      confirm_text = settings.button_label.c_str();
-    else if (action == GTK_FILE_CHOOSER_ACTION_SAVE)
-      confirm_text = gtk_util::kSaveLabel;
-    else if (action == GTK_FILE_CHOOSER_ACTION_OPEN)
-      confirm_text = gtk_util::kOpenLabel;
-
     InitGtkFileChooserNativeSupport();
 
     if (*supports_gtk_file_chooser_native) {
-      dialog_ = GTK_FILE_CHOOSER(
-          dl_gtk_file_chooser_native_new(settings.title.c_str(), NULL, action,
-                                         confirm_text, gtk_util::kCancelLabel));
+      const char* confirm_text = settings.button_label.empty()
+                                     ? nullptr
+                                     : settings.button_label.c_str();
+      dialog_ = GTK_FILE_CHOOSER(dl_gtk_file_chooser_native_new(
+          settings.title.c_str(), NULL, action, confirm_text, nullptr));
     } else {
+      const char* confirm_text = settings.button_label.empty()
```

Comment:
We also need to generate save label for `GTK_FILE_CHOOSER_ACTION_SAVE`, might want to retain the previous conditions inside this block.

```suggestion
    
const char* confirm_text = gtk_util::GetOkLabel();

if (!settings.button_label.empty())
      confirm_text = settings.button_label.c_str();
else if (action == GTK_FILE_CHOOSER_ACTION_SAVE)
      confirm_text = gtk_util::GetSaveLabel();
else if (action == GTK_FILE_CHOOSER_ACTION_OPEN)
      confirm_text = gtk_util::GetOpenLabel();
```